atlasdb-cassandra/src/test/java/com/palantir/atlasdb/keyvalue/cassandra/CassandraClientPoolTest.java

```diff
@@ -110,22 +119,21 @@ public void setup() {
         when(config.getKeyspaceOrThrow()).thenReturn("ks");
         blacklist = new Blacklist(config, Refreshable.only(UNRESPONSIVE_HOST_BACKOFF_SECONDS));
 
-        doAnswer(invocation -> poolServers.add(getInvocationAddress(invocation)))
+        doAnswer(invocation -> poolServers.putIfAbsent(
+                        getCassandraServerFromInvocation(invocation), mock(CassandraClientPoolingContainer.class)))
```

Comment:
you might want to extract out a single mock'd CCPC for ease of use here (unless you do some very fancy thing mocking individual CCPCs)

Reply:
`(unless you do some very fancy thing mocking individual CCPCs)` we do need this unfortunately to tell if we are re-using containers or not :(